### DISCUSSION TITLE
Catch the save event on a word doc?

### DISCUSSION THREAD

**Discussion by NL33** [dolanmiu/docx#955] [Q&A]
I'm trying to be able to know, with nodejs, when a user saves a word doc. Is this possible with docx?

**Comment by dolanmiu** [comment]
Can you elaborate what does "user saves a word doc" means? Do you mean when the code finishes executing? 

You can have a callback method on the `Packer`

All examples in the `/demo` folder has this

**Comment by NL33** [comment]
Thanks for your reply. This is my goal:
--user has a word document open on their computer (not necessarily a doc created by docx--just a normal word doc)
--user hits the save button on the doc
--my electron app can get notified that the save button on the doc has been clicked/the doc has been saved.

Given that I am looking for a way to perform this action on any word doc (and not a doc created with docx), I expect this might not work. But I thought I'd ask anyway as I am trying to see if this is possible.

**Reply by dolanmiu** [reply]
Sorry, this won't work I'm afraid

The docx library's purpose is to solely turn your JS/TS into a word document

You may want to look into maybe a Microsoft Word API? to somehow get the "save hook event" or something similar

Or an easier option is to continuously "watch" the folder the Word document is being saved to, and then when your code notices a difference in the folder (user changed/saved the document), then action accordingly.

Just some suggestions :) 

Thanks for your interest in `docx` anyway.

**Comment by NL33** [comment]
Those are very helpful suggestions. Thank you.

And `docx` is really well done.

**Reply by dolanmiu** [reply]
thank you!
